python/shotgun_desktop/startup.py

```diff
@@ -478,237 +334,118 @@ def __extract_command_line_argument(arg_name):
     return is_set
 
 
-def __launch_app(app, splash, connection, app_bootstrap, server, settings):
+def __launch_app(app, splash, user, app_bootstrap, server, settings):
     """
     Shows the splash screen, optionally downloads and configures Toolkit, imports it, optionally
     updates it and then launches the desktop engine.
 
     :param app: Application object for event processing.
     :param splash: Splash dialog to update user on what is currently going on
-    :param connection: Connection to the Shotgun server.
+    :param user: Current ShotgunUser.
+    :param app_bootstrap: Application bootstrap.
     :param server: The tk_framework_desktopserver.Server instance.
     :param settings: The application's settings.
 
     :returns: The error code to return to the shell.
     """
     # show the splash screen
     splash.show()
+
+    import sgtk
+
+    sgtk.set_authenticated_user(user)
+
+    # Downloads an upgrade for the startup if available. The startup upgrade is independent from the
+    # auto_path state and has its own logic for auto-updating or not, so move this outside the
```

Comment:
is desktop using auto path at all now? That state is gone now that bootstrap exists, right?